.github/workflows/auto-spotless-apply.yml

```diff
@@ -0,0 +1,109 @@
+name: Auto spotless apply
+on:
+  workflow_run:
+    workflows:
+      - "Auto spotless check"
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  apply:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - id: download-patch
+        name: Download patch
+        uses: actions/github-script@v7.0.1
+        with:
+          # this script copied from
+          # https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#using-data-from-the-triggering-workflow
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id
+            });
+            let patchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "patch"
+            })[0];
+            if (!patchArtifact) {
+              core.info('No patch to apply.');
+              return;
+            }
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: patchArtifact.id,
+               archive_format: 'zip'
+            });
+            const fs = require('fs');
+            const path = require('path');
+            const temp = '${{ runner.temp }}/artifacts';
+            if (!fs.existsSync(temp)){
+              fs.mkdirSync(temp);
+            }
+            fs.writeFileSync(path.join(temp, 'patch.zip'), Buffer.from(download.data));
+            core.setOutput("exists", "true");
+
+      - id: get-pr-number
+        name: Get PR number
+        uses: actions/github-script@v7.0.1
+        with:
+          script: |
+            const response = await github.request(context.payload.workflow_run.url);
+            core.setOutput('pr-number', response.data.pull_requests[0].number);
+
+      - name: Unzip patch
+        if: steps.download-patch.outputs.exists == 'true'
+        working-directory: ${{ runner.temp }}/artifacts
+        run: unzip patch.zip
+
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        if: steps.download-patch.outputs.exists == 'true'
```

Comment:
need to use token with write access to push back to the branch, but not the default github token since that won't trigger workflows to re-run